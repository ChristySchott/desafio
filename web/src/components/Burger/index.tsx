import React from 'react';

import Container from './styles';

interface ContainerProps {
  open: boolean;
  setOpen: Function;
}

const Burger: React.FC<ContainerProps> = ({ open, setOpen }) => {
  return (
    <Container open={open} onClick={() => setOpen(!open)}>
      <div />
      <div />
      <div />
    </Container>
  );
};

export default Burger;
