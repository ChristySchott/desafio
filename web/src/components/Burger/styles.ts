import styled, { css } from 'styled-components';

interface ContainerProps {
  open: boolean;
}

const Container = styled.button<ContainerProps>`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 2.4rem;
  height: 2.4rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;

  ${({ open }) =>
    open &&
    css`
      position: absolute;
      top: 4%;
      left: 2.5rem;
      transition: top 1s ease-out;
    `};

  &:focus {
    outline: none;
  }

  div {
    width: 2.5rem;
    height: 0.3rem;
    background-color: ${({ open }) => (open ? '#fff' : '#000')};
    border-radius: 10px;
    transition: transform 0.3s linear;
    position: relative;
    transform-origin: 1px;

    &:first-child {
      transform: ${({ open }) => (open ? 'rotate(45deg)' : 'rotate(0)')};
    }

    &:nth-child(2) {
      opacity: ${({ open }) => (open ? '0' : '1')};
      transform: ${({ open }) => (open ? 'translateX(20px)' : 'translateX(0)')};
    }

    &:nth-child(3) {
      transform: ${({ open }) => (open ? 'rotate(-45deg)' : 'rotate(0)')};
    }
  }
`;

export default Container;
