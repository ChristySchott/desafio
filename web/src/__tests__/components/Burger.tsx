/* eslint-disable no-console */
import React from 'react';
import { render } from '@testing-library/react';

import Burger from '../../components/Burger';

describe('<Burger />', () => {
  it('should render with no error', () => {
    const { container } = render(
      <Burger
        open
        setOpen={() => {
          console.log('Teste');
        }}
      />,
    );

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
