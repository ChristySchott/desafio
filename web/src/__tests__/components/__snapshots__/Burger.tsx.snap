// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Burger /> should render with no error 1`] = `
<button
  class="sc-bdnylx kAubux"
  open=""
>
  <div />
  <div />
  <div />
</button>
`;
